### ISSUE TITLE
Feature Request: Resource for create repository branch

### ISSUE THREAD

**Issue by scarbeau** [gitlabhq/terraform-provider-gitlab#212]
## Use case:
Create new branch on repository

## Example
```hcl
resource "gitlab_branch_creation" "prod_branch" {
  project   = "12345"
  branch    = "production"
  ref       = "master"
}
```

## API Reference
https://docs.gitlab.com/ee/api/branches.html#create-repository-branch

**Comment by Shocktrooper** [comment]
Ay i'm looking for this as well. If someone wants to partner I would like to try to implement but I have not worked on this project before

**Comment by timofurrer** [comment]
This has been implemented in #634 and released with v3.11.0 🎉
